src/Compilers/CSharp/Portable/Lowering/Instrumentation/ModuleCancellationInstrumenter.cs

```diff
@@ -0,0 +1,224 @@
+            if (member == _factory.TopLevelMethod?.OriginalDefinition)
+            {
+                continue;
+            }
+
+            // It's unlikely that real-world APIs have overloads that differ in dynamic,
+            // but if they do avoid selecting them since their intended use might differ.
+            //
+            // Similarly, we also only consider overloads with the same visibility.
+            // We could potentially allow the visibility to differ,
+            // but we'd need to check if the overload is accessible at the call site.
+
+            const TypeCompareKind TypeComparisonKind = TypeCompareKind.CLRSignatureCompareOptions & ~TypeCompareKind.IgnoreDynamic;
+
+            if (member.IsStatic == methodDefinition.IsStatic &&
+                member.MetadataVisibility == methodDefinition.MetadataVisibility &&
+                member is MethodSymbol { Parameters: [.., { RefKind: RefKind.None, Type: { } lastParamType }] parametersWithCancellationToken } overload &&
+                overload.Arity == methodDefinition.Arity &&
+                methodDefinition.Parameters.Length == parametersWithCancellationToken.Length - 1 &&
+                lastParamType.Equals(_throwMethod.ContainingType, TypeCompareKind.ConsiderEverything) &&
+                (!methodsSetsRequiredMembers || overload.HasSetsRequiredMembers))
+            {
+                // we are only considering constructors and ordinary methods, which have different name.
+                Debug.Assert(methodDefinition.MethodKind == overload.MethodKind);
+
+                var typeMap = (methodDefinition.Arity > 0) ? new TypeMap(overload.TypeParameters, methodDefinition.TypeParameters) : null;
+
+                if (MemberSignatureComparer.HaveSameParameterTypes(
+                        methodDefinition.Parameters.AsSpan(),
+                        typeMap1: null,
+                        parametersWithCancellationToken.AsSpan(0, methodDefinition.Parameters.Length),
+                        typeMap,
+                        MemberSignatureComparer.RefKindCompareMode.ConsiderDifferences,
+                        TypeComparisonKind) &&
+                    MemberSignatureComparer.HaveSameReturnTypes(
+                        methodDefinition,
+                        typeMap1: null,
+                        overload,
+                        typeMap,
+                        TypeComparisonKind) &&
+                    MemberSignatureComparer.HaveSameConstraints(
+                        methodDefinition.TypeParameters,
+                        typeMap1: null,
+                        overload.TypeParameters,
+                        typeMap))
+                {
+                    var result = overload.AsMember(method.ContainingType);
+                    return (result.Arity > 0) ? result.Construct(method.TypeArgumentsWithAnnotations) : result;
```

Comment:
>return [](http://example.com/codeflow?start=20&length=6)

It looks like we are not worried about an ambiguity (can happen in metadata due to custom modifier differences). That is probably fine, but it would be good to state this explicitly in a comment.

Reply:
> It looks like we are not worried about an ambiguity

Never mind, we are using ```CLRSignatureCompareOptions``` for comparison